src/figuresClasses.ts

```diff
@@ -1,19 +1,88 @@
+export enum Shape {
+  Triangle = 'triangle',
+  Circle = 'circle',
+  Rectangle = 'rectangle'
+}
+
+export type Color = 'red' | 'green' | 'blue';
+
 export interface Figure {
+  shape: Shape;
+  color: Color;
+  getArea(): number;
+}
 
+export function roundTo2(area: number): number {
+  return Math.floor(area * 100) / 100;
 }
 
-export class Triangle {
+export class Triangle implements Figure {
+  shape = Shape.Triangle;
 
-}
+  getArea(): number {
+    const semiperimeter: number = (this.a + this.b + this.c) / 2;
+
+    const area: number
+    = Math.sqrt(semiperimeter
+    * (semiperimeter - this.a)
+    * (semiperimeter - this.b)
+    * (semiperimeter - this.c));
 
-export class Circle {
+    return roundTo2(area);
+  }
 
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (a <= 0 || b <= 0 || c <= 0
```

Comment:
If you want, you can create separate function for checking figure sides (if they <= 0) and call this function with the appropriate parameters. Because this checking repeat in all classes.